packages/react/src/progress-indicator/ProgressIndicator.tsx

```diff
@@ -34,49 +51,82 @@ export const ProgressIndicator = ({
 				items.length
 		  } steps completed`;
 
-	const hasExplicitActiveDeclared =
-		items.filter(({ isActive }) => isActive).length > 0;
+	/* **LEGACY HANDLER**
+	 * This is now legacy, please keep this for backwards compatibility.
+	 */
+	const hasExplicitActiveDeclared = items.some(({ isActive }) => isActive);
+
+	if (process.env.NODE_ENV !== 'production' && hasExplicitActiveDeclared) {
+		console.warn(
+			'ProgressIndicator: The `isActive` key is deprecated. Please use `activePath` in the parent component, `<ProgressIndicator />`.'
+		);
+	}
+
+	/* **LEGACY HANDLER**
+	 * NOTE: This is for handling legacy props. We deprecated status='doing' in
+	 * favour of status='started' and this code applies `isActive` to the legacy 'doing'
+	 * items, enabling backwards compatibility. We then deprecated the `isActive` prop
+	 * in favour of `activePath` to help manage level two active items and for
+	 * consistency across other nav list components.
+	 */
+	const itemsWithDefaultActive = hasExplicitActiveDeclared
+		? items
+		: items.map((item) => {
+				/* **LEGACY HANDLER**
+				 * If no explicit active items are declared, the 'doing' status item receives active state by default.
+				 * NOTE: this is important to keep for backwards compatibility.
+				 */
+				const isActiveFromLegacyDoingStatus = item.status === 'doing';
+
+				const activePathMatcher = 'href' in item ? item.href : item.label;
```

Comment:
Does this also work?
```suggestion
				const activePathMatcher = item?.href || item.label;
```

Reply:
Not with TypeScript 😢

Reply:
Urgh punishing users with more code to satisfy a compiler is my most hated of things about TS